go-controller/pkg/ovn/routeimport/route_import.go

```diff
@@ -0,0 +1,560 @@
+	}
+
+	c.reconciler = controllerutil.NewReconciler(
+		"RouteImport",
+		&controllerutil.ReconcilerConfig{
+			Threadiness: 1,
+			Reconcile:   c.syncNetwork,
+			RateLimiter: workqueue.NewTypedItemFastSlowRateLimiter[string](time.Second, 5*time.Second, 5),
+		},
+	)
+
+	return c
+}
+
+type netInfo struct {
+	util.NetInfo
+	id    int
+	table int
+}
+
+type controller struct {
+	ctx        util.CancelableContext
+	nbClient   client.Client
+	node       string
+	log        logr.Logger
+	reconciler controllerutil.Reconciler
+	netlink    util.NetLinkOps
+
+	sync.RWMutex
+	networkIDs map[int]string
+	networks   map[string]*netInfo
+	tables     map[int]int
+}
+
+func (c *controller) AddNetwork(network util.NetInfo) error {
+	c.Lock()
+	defer c.Unlock()
+
+	networkID := network.GetNetworkID()
+	if c.networkIDs[networkID] != "" {
+		return fmt.Errorf("already tracking network %q with ID %d",
+			c.networkIDs[networkID],
+			networkID,
+		)
+	}
+
+	name := network.GetNetworkName()
+	if c.networks[name] != nil {
+		return fmt.Errorf("already tracking network name %q", name)
+	}
+
+	info := &netInfo{NetInfo: network, id: networkID}
+	if network.IsDefault() {
+		c.tables[unix.RT_TABLE_MAIN] = networkID
+		info.table = unix.RT_TABLE_MAIN
+	}
+	c.networkIDs[networkID] = name
+	c.networks[name] = info
+
+	c.log.V(5).Info("Added network", "name", name, "id", networkID)
+	c.reconcile(name)
+
+	return nil
+}
+
+func (c *controller) ForgetNetwork(name string) {
+	c.Lock()
+	defer c.Unlock()
+
+	network := c.networks[name]
+	if network == nil {
+		return
+	}
+
+	delete(c.networkIDs, network.id)
+	delete(c.networks, name)
+
+	c.log.V(5).Info("Forgot network", "name", name)
+}
+
+func (c *controller) NeedsReconciliation(network util.NetInfo) bool {
+	c.RLock()
+	defer c.RUnlock()
+
+	if c.networks[network.GetNetworkName()] == nil {
+		return false
+	}
+
+	// TODO check if overlay mode changed
+	return false
+}
+
+func (c *controller) ReconcileNetwork(name string) error {
+	c.RLock()
+	defer c.RUnlock()
+	if c.networks[name] == nil {
+		return fmt.Errorf("unknown network with name %q", name)
+	}
+	c.log.V(5).Info("Reconciling network", "name", name)
+	c.reconcile(name)
+	return nil
+}
+
+func (c *controller) Start() error {
+	defer c.log.Info("Controller started")
+	c.subscribe(c.ctx.Done())
+	return controllerutil.Start(c.reconciler)
+}
+
+func (c *controller) Stop() {
+	controllerutil.Stop(c.reconciler)
+	c.ctx.Cancel()
+	c.log.Info("Controller stopped")
+}
+
+func (c *controller) subscribe(stop <-chan struct{}) {
+	setTimer := func(t *time.Timer, d time.Duration) *time.Timer {
+		if t == nil {
+			return time.NewTimer(d)
+		}
+		t.Stop()
+		select {
+		case <-t.C:
+		default:
+		}
+		t.Reset(d)
+		return t
+	}
+
+	go func() {
+		onError := func(err error) {
+			c.log.Error(err, "Error on netlink route event subscription")
+		}
+		routeEventCh := subscribeNetlinkRouteEvents(c.netlink, stop, onError)
+		t := setTimer(nil, subscribePeriod)
```

Comment:
if we stop we need to in this goroutine we need to stop the timer to avoid leak. In go 1.23 we dont have to but we are still on 1.22 I think:

https://pkg.go.dev/time#NewTimer

Reply:
Technically not a leak per se, it's just that the timer is not garbage collected until it expires. This becomes a problem if you are spawning a lot of different timers with long timeouts or in shorts periods of time (using constructs as `time.After` or the like). Here we are just reusing the same timers all over again, so its only two.

Anyway, aside from the clarification, you are right.